### ISSUE TITLE
Examples doesn't work

### ISSUE THREAD

**Issue by penguinlav** [danields761/pydantic-settings#1]
Idea of project is great!
But I could not run it in trivial cases. For example even examples from [doc](https://pydantic-settings.readthedocs.io/en/latest/) doesn't work.
```
~$ cat example.py
from pydantic import BaseModel

from pydantic_settings import BaseSettingsModel, load_settings


class ComponentOptions(BaseModel):
    val: str


class AppSettings(BaseSettingsModel):
    class Config:
        env_prefix = 'FOO'

    component: ComponentOptions


assert load_settings(
    AppSettings,
    '{}',
    load_env=True,
    type_hint='json',
    _environ={'FOO_COMPONENT_VAL': 'SOME VALUE'}
).component.val == 'SOME VALUE'
```
```
~$ python -m example.py
...
lib/python3.6/site-packages/pydantic_settings/utils.py", line 13, in deep_merge_mappings
    keys = set(first_map).union(set(second_map))
TypeError: 'NoneType' object is not iterable
```

**Comment by danields761** [comment]
Hello!

Thanks for good tip and reporting this issue!

I haven't invented any effort in this project for a long time and there is big *pydantic* changelog. I'am trying to figure out this and advance project. So anyway this issue will be fixed soon.

**Comment by danields761** [comment]
Hello!

This issue has been fixed in latest release (0.2.5)
